### PR TITLE
ICU-21323 Automates BRS testing tasks of U_CHARSET_IS_UTF8 and

### PR DESCRIPTION
**Pull request by gnrunge**
U_OVERRIDE_CXX_ALLOCATION=0.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21323
- [x] Required: The PR title must be prefixed with a JIRA Issue number. For example: "ICU-1234 Fix xyz"
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. For example: "ICU-1234 Fix xyz"
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: .github/workflows/icu_ci.yml [hunk 1]
```diff
@@ -191,6 +191,27 @@ jobs:
...
+          make -j2 check
+
+  # Test U_OVERRIDE_CXX_ALLOCATION-is-0-test
+  u-override-cxx-allocation-is-0-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          cd icu4c/source;
+          ./runConfigureICU Linux CPPFLAGS="-DU_OVERRIDE_CXX_ALLOCATION=0";
```
Comment (jefgen) [inline .github/workflows/icu_ci.yml:211]: Curiously the doc page for this test has the variable set before calling `./runConfigureICU`: 
http://site.icu-project.org/processes/release/tasks/healthy-code#TOC-Test-U_OVERRIDE_CXX_ALLOCATION-0

`CPPFLAGS="-DU_OVERRIDE_CXX_ALLOCATION=0" ./runConfigureICU Linux`
vs
`./runConfigureICU Linux CPPFLAGS="-DU_OVERRIDE_CXX_ALLOCATION=0";

Though perhaps it doesn't matter for this test.

Comment (markusicu) [reply .github/workflows/icu_ci.yml:211]: I assume that for environment variables, they have to come before the command, and that if you put them after, then the command sees them as garbage options. No? (FYI @srl295)

Comment (srl295) [reply .github/workflows/icu_ci.yml:211]: > I assume that for environment variables, they have to come before the command, and that if you put them after, then the command sees them as garbage options. No? (FYI @srl295)

No, you can set variables afterwards this way.

It's a feature of `configure`.  Search for `*=*)` (as a literal, not a regex) in `configure`, or `configure --help`

> To assign environment variables ... specify them as VAR=VALUE.

Comment (gnrunge) [reply .github/workflows/icu_ci.yml:211]: It does not matter. I ran runConfigureICU either way and the diffs were limited to the order of the flags in config.status in a few lines. The generated Makefiles have no differences and show the CPP flag at the compiler invokation.

I put the CPPFLAGS after the runConfigureICU to make it uniform with the U_CHARSET_IS_UTF8 test a few lines earlier.

Comment (jefgen) [reply .github/workflows/icu_ci.yml:211]: Thanks @srl295 and @gnrunge!

**Review by jefgen — APPROVED** [review]
lgtm, thanks

**Pull request merged**